protocol-units/bridge/contracts/src/AtomicBridgeInitator.sol

```diff
@@ -0,0 +1,137 @@
+import {console} from "forge-std/Test.sol";
+import "./IAtomicBridgeInitiator.sol";
+import "./WETH/interfaces/IWETH10.sol";
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.6;
+
+contract AtomicBridgeInitiator is IAtomicBridgeInitiator {
+    struct BridgeTransfer {
+        uint256 amount;
+        address originator;
+        address recipient;
+        bytes32 hashLock;
+        uint timeLock;
+        bool exists;
+    }
+
+    mapping(bytes32 => BridgeTransfer) public bridgeTransfers;
+    mapping(bytes32 => BridgeTransfer) public completedBridgeTransfers;
+    IWETH10 public weth;
+
+    constructor(address _weth) {
+        weth = IWETH10(_weth);
+    }
+
+    function initiateBridgeTransfer(
```

Comment:
we might want to check these params for validity to avoid accidental locking of funds...

Reply:
wouldn't the recipient but on the destination chain, which in our case is Movement, how would this type `address` translate exactly?

Reply:
Yes this was a todo in my mind. Would have to be a hardcoded string

Reply:
Does this mean it will be resolved now?